File: src/main/resources/output/IFC4X3_csharp/Entities/IfcProfileDef/IfcArbitraryClosedProfileDef/IfcArbitraryProfileDefWithVoids/IfcArbitraryProfileDefWithVoids.cs
public class IfcArbitraryProfileDefWithVoids : IfcArbitraryClosedProfileDef
{
    public SET [1:?] OF IfcCurve InnerCurves { get; set; }

    // === WHERE CLAUSES ===
    // WR1 : SELF\IfcProfileDef.ProfileType = AREA
    // WR2 : SIZEOF(QUERY(temp <* InnerCurves | temp.Dim <> 2)) = 0
    // WR3 : SIZEOF(QUERY(temp <* InnerCurves | 'IFC4X3_DEV_73740fe4.IFCLINE' IN TYPEOF(temp))) = 0
}

File: output/ifc4x3_dev_73740fe4/entity/IfcArbitraryProfileDefWithVoids.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.Set;

public class IfcArbitraryProfileDefWithVoids extends IfcArbitraryClosedProfileDef {

    // INVERSE attributes:
    // WR1 : SELF\IfcProfileDef.ProfileType = AREA;
    // WR2 : SIZEOF(QUERY(temp <* InnerCurves | temp.Dim <> 2)) = 0;
    // WR3 : SIZEOF(QUERY(temp <* InnerCurves | 'IFC4X3_DEV_73740fe4.IFCLINE' IN TYPEOF(temp))) = 0;

    // WHERE constraints:
    // WR1 : SELF\IfcProfileDef.ProfileType = AREA;
    // WR2 : SIZEOF(QUERY(temp <* InnerCurves | temp.Dim <> 2)) = 0;
    // WR3 : SIZEOF(QUERY(temp <* InnerCurves | 'IFC4X3_DEV_73740fe4.IFCLINE' IN TYPEOF(temp))) = 0;
}
